### ISSUE TITLE
Task runner not working with tcsh and csh shells

### ISSUE THREAD

**Issue by danieltomasku** [eclipse-theia/theia#8164]
### Bug Description:
Launching a task while using `tcsh` or `csh`  shells does not work and exits immediately. This is due to the startup flags being passed to it, `-l` and `-c`. According to the `tcsh` [documentation](https://linux.die.net/man/1/tcsh), the `-l` flag is valid if it is the _only_ flag being passed, so `tcsh -l -c` will not work.

Relevant code area;
https://github.com/eclipse-theia/theia/blob/a1ea2307d50565944b42f74ba3797698697ec897/packages/task/src/node/process/process-task-runner.ts#L178-L197

Seems like a conditional block can be added to the above code snippet that matches on (t)csh and then only passes `-c`.

### Steps to Reproduce:

1. If you have `tcsh` available, set the `THEIA_SHELL` env variable to your tcsh location, e.g. `export THEIA_SHELL=/bin/tcsh`
2. Run Theia
3. Create a task in `tasks.json` of type `shell`.
4. Try running that task and confirm it exits immediately with code 1

### Additional Information

- Operating System: Red Hat Enterprise Linux 7
- Theia Version: 1.3.0

**Comment by akosyakov** [comment]
@marechal-p a bug?

**Comment by marechal-p** [comment]
@akosyakov just an unsupported shell. @danieltomasku is your shell supported when running tasks with VS Code?

**Comment by danieltomasku** [comment]
@marechal-p yes, the `tcsh` shell is supported when running tasks with VS Code:
![image](https://user-images.githubusercontent.com/14880569/87444136-02d10e00-c5bc-11ea-922a-36621321bbdf.png)
